client/extensions/woocommerce/app/settings/payments/payments-off-site.js

```diff
@@ -9,27 +9,89 @@ import { localize } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
+import {
+	changePaymentMethodField,
+	closeEditingPaymentMethod,
+	openPaymentMethodForEdit
+} from 'woocommerce/state/ui/payments/methods/actions';
+import { errorNotice, successNotice } from 'state/notices/actions';
 import ExtendedHeader from 'woocommerce/components/extended-header';
-import { fetchPaymentMethods } from 'woocommerce/state/sites/payment-methods/actions';
-import { getPaymentMethodsGroup } from 'woocommerce/state/sites/payment-methods/selectors';
+import { fetchPaymentMethods, paymentMethodSave } from 'woocommerce/state/sites/payment-methods/actions';
+import { getCurrentlyEditingPaymentMethod, getPaymentMethodsGroup } from 'woocommerce/state/ui/payments/methods/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
 import List from 'woocommerce/components/list/list';
 import ListHeader from 'woocommerce/components/list/list-header';
 import ListItemField from 'woocommerce/components/list/list-item-field';
+import PaymentMethodEdit from './payment-method-edit';
 import PaymentMethodItem from './payment-method-item';
 
 class SettingsPaymentsOffSite extends Component {
 	static propTypes = {
-		paymentMethods: PropTypes.array,
+		closeEditingPaymentMethod: PropTypes.func.isRequired,
+		currentlyEditingMethod: PropTypes.object,
 		fetchPaymentMethods: PropTypes.func.isRequired,
+		openPaymentMethodForEdit: PropTypes.func.isRequired,
+		paymentMethods: PropTypes.array,
+		paymentMethodSave: PropTypes.func,
+		siteId: PropTypes.number,
 	};
 
 	componentDidMount() {
 		this.props.fetchPaymentMethods();
 	}
 
+	onCancel = ( method ) => {
+		const { siteId } = this.props;
+		this.props.closeEditingPaymentMethod( siteId, method.id );
```

Comment:
Optional - please consider renaming these event handlers to something like 'on...' (e.g. onCloseEditingPaymentMethod) to make it more consistent with other event handlers in calypso